File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task133_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <string>

using namespace std;

void resetPassword() {
    string currentPassword, newPassword, confirmPassword;

    while (true) {
        cout << "Enter your current password: ";
        getline(cin, currentPassword);
        if (currentPassword == "password123") {
            break;
        } else {
            cout << "Incorrect password. Please try again." << endl;
        }
    }

    while (true) {
        cout << "Enter your new password: ";
        getline(cin, newPassword);
        cout << "Confirm your new password: ";
        getline(cin, confirmPassword);
        if (newPassword == confirmPassword) {
            cout << "Password reset successful." << endl;
            break;
        } else {
            cout << "Passwords don't match. Please try again." << endl;
        }
    }
}

int main() {
    resetPassword();
    return 0;
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task133_GEMINI_gemini-1.5-pro-001.java
import java.util.Scanner;

public class PasswordReset {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        resetPassword(scanner);
    }

    public static void resetPassword(Scanner scanner) {
        while (true) {
            System.out.print("Enter your current password: ");
            String currentPassword = scanner.nextLine();
            if (currentPassword.equals("password123")) {
                break;
            } else {
                System.out.println("Incorrect password. Please try again.");
            }
        }

        while (true) {
            System.out.print("Enter your new password: ");
            String newPassword = scanner.nextLine();
            System.out.print("Confirm your new password: ");
            String confirmPassword = scanner.nextLine();
            if (newPassword.equals(confirmPassword)) {
                System.out.println("Password reset successful.");
                break;
            } else {
                System.out.println("Passwords don't match. Please try again.");
            }
        }
    }
}